internal/resources/asserts/resource_alert_config.go

```diff
@@ -0,0 +1,291 @@
+		schema,
+	).WithLister(assertsListerFunction(listAlertConfigs))
+}
+
+func resourceAlertConfigCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*common.Client).AssertsAPIClient
+	if client == nil {
+		return diag.Errorf("Asserts API client is not configured")
+	}
+
+	stackID := meta.(*common.Client).GrafanaStackID
+	if stackID == 0 {
+		return diag.Errorf("stack_id must be set in provider configuration for Asserts resources")
+	}
+	name := d.Get("name").(string)
+	matchLabels := make(map[string]string)
+	alertLabels := make(map[string]string)
+
+	if v, ok := d.GetOk("match_labels"); ok {
+		for k, val := range v.(map[string]interface{}) {
+			matchLabels[k] = val.(string)
+		}
+	}
+
+	if v, ok := d.GetOk("alert_labels"); ok {
+		for k, val := range v.(map[string]interface{}) {
+			alertLabels[k] = val.(string)
+		}
+	}
+
+	duration := d.Get("duration").(string)
+	silenced := d.Get("silenced").(bool)
+
+	// Create AlertConfigDto using the generated client models
+	// Only include non-empty maps to avoid validation issues
+	alertConfig := assertsapi.AlertConfigDto{
+		Name:     &name,
+		Silenced: &silenced,
+	}
+
+	// Only set matchLabels if not empty
+	if len(matchLabels) > 0 {
+		alertConfig.MatchLabels = matchLabels
+	}
+
+	// Only set alertLabels if not empty
+	if len(alertLabels) > 0 {
+		alertConfig.AlertLabels = alertLabels
+	}
+
+	// Only set duration if not empty
+	if duration != "" {
+		alertConfig.For = &duration
+	}
+
+	// Call the generated client API
+	request := client.AlertConfigurationAPI.PutAlertConfig(ctx).
+		AlertConfigDto(alertConfig).
+		XScopeOrgID(fmt.Sprintf("%d", stackID))
+
+	_, err := request.Execute()
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("failed to create alert configuration: %w", err))
+	}
+
+	d.SetId(name)
+	return resourceAlertConfigRead(ctx, d, meta)
+}
+
+func resourceAlertConfigRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*common.Client).AssertsAPIClient
+	if client == nil {
+		return diag.Errorf("Asserts API client is not configured")
+	}
+
+	stackID := meta.(*common.Client).GrafanaStackID
+	if stackID == 0 {
+		return diag.Errorf("stack_id must be set in provider configuration for Asserts resources")
+	}
+	name := d.Id()
+
+	// Get all alert configs using the generated client API
+	request := client.AlertConfigurationAPI.GetAllAlertConfigs(ctx).
+		XScopeOrgID(fmt.Sprintf("%d", stackID))
+
+	alertConfigs, _, err := request.Execute()
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("failed to get alert configurations: %w", err))
+	}
+
+	// Find our specific config
+	var foundConfig *assertsapi.AlertConfigDto
+	for _, config := range alertConfigs.AlertConfigs {
+		if config.Name != nil && *config.Name == name {
+			foundConfig = &config
+			break
+		}
+	}
+
+	if foundConfig == nil {
+		// Resource not found - this could be due to eventual consistency
```

Comment:
This will almost always fail, maybe we can have a little waiting and retry logic to prevent consistent failures. This should be available for reads within 1 min or so.